src/downloader.rs

```diff
@@ -91,20 +92,58 @@ impl Downloader {
         main: Arc<ProgressBar>,
     ) -> Summary {
         // Create a download summary.
-        let mut summary = Summary::new(download.clone(), StatusCode::BAD_REQUEST, 0);
-
-        // Check if we should skip the file or continue.
+        let mut size_on_disk: u64 = 0;
+        let mut resumable = false;
+        let mut summary = Summary::new(
+            download.clone(),
+            StatusCode::BAD_REQUEST,
+            size_on_disk,
+            resumable,
+        );
         let output = self.directory.join(&download.filename);
-        if self.skip_existing && output.exists() {
-            return summary.with_status(Status::Skipped(format!(
-                "a file with the same name already exists at the destination: {:?}",
-                &output
-            )));
+
+        // If resumable is turned on...
+        if self.resumable {
+            // Check whether the download is resumable.
+            let res = match client.head(download.url.clone()).send().await {
+                Ok(res) => res,
+                Err(e) => {
+                    return summary.fail(e);
+                }
+            };
+            let headers = res.headers();
+            resumable = match headers.get(ACCEPT_RANGES) {
+                None => false,
+                Some(x) if x == "none" => false,
+                Some(_) => true,
+            };
+
+            // If we can resume...
+            if resumable {
```

Comment:
should these two be combined: `if resumable && output.exists()`?

Reply:
Yes, good point. I can probably add some clarity to this block.